haystack/preview/components/embedders/openai_document_embedder.py

```diff
@@ -0,0 +1,163 @@
+from typing import List, Optional, Dict, Any, Tuple
+import os
+
+import openai
+from tqdm import tqdm
+
+
+from haystack.preview import component, Document, default_to_dict, default_from_dict
+
+
+@component
+class OpenAIDocumentEmbedder:
+    """
+    A component for computing Document embeddings using OpenAI models.
+    The embedding of each Document is stored in the `embedding` field of the Document.
+    """
+
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        model_name: str = "text-embedding-ada-002",
+        organization: Optional[str] = None,
+        prefix: str = "",
+        suffix: str = "",
+        batch_size: int = 32,
+        progress_bar: bool = True,
+        metadata_fields_to_embed: Optional[List[str]] = None,
+        embedding_separator: str = "\n",
+    ):
+        """
+        Create a OpenAIDocumentEmbedder component.
+        :param api_key: The OpenAI API key.
```

Comment:
We should mention here that the API key is taken from the env variable "OPENAI_API_KEY" by default.